rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -182,11 +182,13 @@ impl SerialSubmitter {
         // queue for further processing.
 
         // Promote any newly-ready messages from the wait queue to the run queue.
-        let wait_messages: Vec<_> = self.wait_queue.drain(..).collect();
+        // The order of wait_messages is preserved and pushed at the front of the run_queue
+        // to ensure that new messages are evaluated first.
+        let wait_messages: Vec<_> = self.wait_queue.drain(..).rev().collect();
         for msg in wait_messages {
             // TODO(webbhorn): Check against interchain gas paymaster.  If now enough payment,
             // promote to run queue.
-            self.run_queue.push(msg);
+            self.run_queue.push_front(msg);
         }
         self.wait_queue = Vec::new();
```

Comment:
This also does not make sense to me, why alloc new memory? We just drained it. I realize you did not change it but noticing it now.

Reply:
Option two is to `.into_iter` above instead of draining and that would at least make a little more sense than drain and recreate, but drain and keep using makes even more sense.

Reply:
agreed this is weird, fixed